cluster/manifests/kubernetes-lifecycle-metrics/deployment.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetes-lifecycle-metrics
+  namespace: kube-system
+  labels:
+    application: kubernetes-lifecycle-metrics
+    version: master-1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      application: kubernetes-lifecycle-metrics
+  template:
+    metadata:
+      labels:
+        application: kubernetes-lifecycle-metrics
+        version: master-1
+      annotations:
+        kubernetes-log-watcher/scalyr-parser: '[{"container": "kubernetes-lifecycle-metrics", "parser": "json"}]'
+    spec:
+      priorityClassName: system-cluster-critical
```

Comment:
Do we really need this for all components that only provide metrics?
IMHO that should not be done, but more for the important things like metric server that cause an outage of HPA or skipper that cause an outage of http routing

Reply:
Why? I don't see why the users' pods should be able to preempt infrastructure stuff.

Reply:
If we have only limited CPU/memory and we have to select the important and the unimportant workloads, then I would say skipper-ingress or Prometheus is more important than this. I think we should use different priorities for different workloads, not treat all the same.

Reply:
It does, but I don't think sure it's *that* important. We always have the users' pods to sacrifice, and it won't kick out anything important, so I see no harm in leaving it as is.